server/text_generation_server/inference_engine/hf_optimum_ov.py

```diff
@@ -0,0 +1,97 @@
+"""
+OpenVINO integration for text-generation-inference.
+
+Usage: set DEPLOYMENT_FRAMEWORK environment variable to hf_optimum_ov or use the
+`--deployment-framework=hf_optimum_ov` flag with text-generation-launcher.
+
+Example, including model conversion:
+
+```
+pip install optimum[openvino,nncf]
+optimum-cli export openvino -m facebook/opt-2.7b data/opt-2.7b-ov
+volume=$PWD/data
+mkdir $volume
+MODEL=/data/opt-2.7b-ov
+IMAGE_ID= [text-generation-server IMAGE_ID]
+docker run -p 8033:8033 -p 3000:3000 -e OPENVINO_CONFIG=/data/openvino_config.json \
+-v $volume:/data $IMAGE_ID text-generation-launcher --model-name $MODEL --deployment-framework hf_optimum_ov
+
+```
+
+OPENVINO_CONFIG is optional. If used, it must point to a json file with an openvino configuration.
+If no specific config is specified, it is set to:
+{"CACHE_DIR": "", "PERFORMANCE_HINT": "LATENCY", "PERFORMANCE_HINT_NUM_REQUESTS": 1}.
+
+Known limitations:
+
+- Seq2Seq models are not supported yet in this integration.
+- Only CPU device is supported at the moment.
+
+"""
+
+import json
+import os
+from pathlib import Path
+from typing import Any, Optional, Union
+
+import torch
+from openvino.runtime import get_version
+from optimum.intel import OVModelForCausalLM
+from optimum.intel.version import __version__
+from text_generation_server.inference_engine.engine import BaseInferenceEngine
+from text_generation_server.utils.hub import TRUST_REMOTE_CODE
+from transformers import AutoModelForCausalLM, AutoModelForSeq2SeqLM
+
+
+class InferenceEngine(BaseInferenceEngine):
+    def __init__(
+        self,
+        model_path: str,
+        model_class: Union[AutoModelForCausalLM, AutoModelForSeq2SeqLM],
+        dtype: torch.dtype,
+        quantize: Optional[str],  # not used by OpenVINO
```

Comment:
Why not to consider `quantize` parameter as a trigger to compress the model weights to INT8 or INT4?

Reply:
`quantize` is currently used for bitsandbytes and GPTQ and using anything else throws an error. We could presumably modify that, but for weight compression it seemed load_in_8bit (which is now the default) and soon load_in_4bit would be a better fit. 

Personally I would always compress offline and load the compressed model directly. TGIS requires downloaded weights. So if you want to compress the model on the fly, you would have to download the full precision weights, keep them on disk, and then within TGIS compress the model to 4 or 8 bit every time, which takes several minutes.

Reply:
Thanks @helena-intel. I agree that offline compression is better. But I noticed you allow on-fly conversion here based only logic below where you add a flag `kwargs["export"] = True` when it is not `model_is_ov`. That is why I ask about on-fly compression for such models as well.

Reply:
It's a good point!  Note that at the moment we already do on-the-fly compression for models with more than 1B parameters because they will be converted to 8-bit by optimum-intel. But it would be good to allow to configure this, especially now that we'll have load_in_4bit in optimum-intel soon. How do you propose to include this? Add a "weight_compression" option for quantize in addition to bitsandbytes and gptq? Or weight_compression_int4 and weight_compression_int8? Currently setting dtype_str to int8 also enables bitsandbytes quantization, so I thought we could use the same, but that doesn't out of the box allow int4 because dtype-str is limited to torch dtypes. That can all be changed, but I would like to get a maintainer's opinion on the best way to do this first.

Another option could be to add an environment variable OPENVINO_WEIGHT_FORMAT and allow specifying an exact config for sym/asym, group size and ratio. Which is the most flexible, but a different API than other inference engines.